### ISSUE TITLE
`VersionedXcm` is not created with `versioned_type`

### ISSUE THREAD

**Issue by franciscoaguirre** [paritytech/polkadot-sdk#3846]
The `VersionedXcm` type is defined like so:

```rust
pub enum VersionedXcm<RuntimeCall> {
	#[codec(index = 2)]
	V2(v2::Xcm<RuntimeCall>),
	#[codec(index = 3)]
	V3(v3::Xcm<RuntimeCall>),
	#[codec(index = 4)]
	V4(v4::Xcm<RuntimeCall>),
}
```

meanwhile, all other versioned types use the `versioned_type` macro to get some automatic functionality.

Is there a reason for this? Should we just use the macro for this type as well?

**Comment by bkontur** [comment]
My guess is that it's because of the type parameter `RuntimeCall` - `pub enum VersionedXcm<RuntimeCall>`, and it seems like the only difference in the derive stuff is `#[scale_info(bounds(), skip_type_params(RuntimeCall))]`.

**Comment by franciscoaguirre** [comment]
True. If there was a way to put that inside of the macro I think we'd save ourselves from some duplication.
